tests/test_routing.py

```diff
@@ -197,6 +197,20 @@ def test_defaults():
     assert adapter.build('foo', {'page': 2}) == '/foo/2'
 
 
+def test_negative():
+    map = r.Map([
+        r.Rule('/foo/<int(signed=True):page>', endpoint='foo'),
+        r.Rule('/bar/<float(signed=True):page>', endpoint='bar')
```

Comment:
I'd also like to have a test that ensures signed values are rejected in unsigned mode